pandas/tests/frame/methods/test_replace.py

```diff
@@ -1308,7 +1308,8 @@ def test_categorical_replace_with_dict(self, replace_dict, final_data):
         expected["b"] = expected["b"].cat.set_categories([1, 2, 3])
         result = df.replace(replace_dict, 3)
         tm.assert_frame_equal(result, expected)
-        with pytest.raises(AssertionError):
+        msg = 'Attributes of DataFrame.iloc[:, 0] (column name="a") are different'
```

Comment:
I am not sure we should test this message. We are not testing `assert_frame_equal` and the message it raises, we are just asserting that df and expected are not equal.

Reply:
Then should I remove that assertion instead?

Reply:
> I am not sure we should test this message. We are not testing `assert_frame_equal` and the message it raises, we are just asserting that df and expected are not equal.

by testing the message, or maybe part of it, we could also partially test how the frames are different.

However, if that is an important part of the test, then maybe the test should check the differences explicitly.

so it does make sense not to test the message.

for now (this PR) maybe could just remove the message check.

maybe an assert_frame_not_equal test helper would be helpful. (not in this PR)

Reply:
Done, I have removed the message check for this specific pytest.raises check.

Reply:
@MomIsBestFriend can we make the pytest.raises message checks for AssertionError optional in the validation script?